arbor/include/arbor/serdes.hpp

```diff
@@ -143,205 +147,219 @@ struct serializer {
-ARB_ARBOR_API void serialize(serializer& ser, const K& k, unsigned long v) {
-    ser.write(to_serdes_key(k), static_cast<unsigned long long>(v));
+ARB_ARBOR_API void serialize(::arb::serializer& ser, const K& k, unsigned long v) {
+    ser.write(arb::to_serdes_key(k), static_cast<unsigned long long>(v));
 }
 
 template<typename K>
-ARB_ARBOR_API void serialize(serializer& ser, const K& k, const float v) {
-    ser.write(to_serdes_key(k), double{v});
+ARB_ARBOR_API void serialize(::arb::serializer& ser, const K& k, const float& v) {
+    ser.write(arb::to_serdes_key(k), double{v});
 }
 
 template<typename K>
-ARB_ARBOR_API void serialize(serializer& ser, const K& k, const double v) {
-    ser.write(to_serdes_key(k), v);
+ARB_ARBOR_API void serialize(::arb::serializer& ser, const K& k, const double& v) {
+    ser.write(arb::to_serdes_key(k), v);
 }
 
 template<typename K>
-ARB_ARBOR_API void serialize(serializer& ser, const K& k, const bool v) {
-    ser.write(to_serdes_key(k), static_cast<long long>(v));
+ARB_ARBOR_API void serialize(::arb::serializer& ser, const K& k, const bool v) {
+    ser.write(arb::to_serdes_key(k), static_cast<long long>(v));
 }
 
 template<typename K>
-ARB_ARBOR_API void serialize(serializer& ser, const K& k, const unsigned long long v) {
-    ser.write(to_serdes_key(k), v);
+ARB_ARBOR_API void serialize(::arb::serializer& ser, const K& k, const unsigned long long v) {
+    ser.write(arb::to_serdes_key(k), v);
 }
 
 template <typename K,
           typename Q,
           typename V>
-ARB_ARBOR_API void serialize(serializer& ser, const K& k, const std::unordered_map<Q, V>& v) {
-    ser.begin_write_map(to_serdes_key(k));
+ARB_ARBOR_API void serialize(::arb::serializer& ser, const K& k, const std::unordered_map<Q, V>& v) {
+    ser.begin_write_map(arb::to_serdes_key(k));
     for (const auto& [q, w]: v) serialize(ser, q, w);
     ser.end_write_map();
 }
 
 template <typename K,
           typename Q,
           typename V>
-ARB_ARBOR_API void serialize(serializer& ser, const K& k, const std::map<Q, V>& v) {
-    ser.begin_write_map(to_serdes_key(k));
+ARB_ARBOR_API void serialize(::arb::serializer& ser, const K& k, const std::map<Q, V>& v) {
+    ser.begin_write_map(arb::to_serdes_key(k));
     for (const auto& [q, w]: v) serialize(ser, q, w);
     ser.end_write_map();
 }
 
 template <typename K,
           typename V,
           typename A>
-ARB_ARBOR_API void serialize(serializer& ser, const K& k, const std::vector<V, A>& vs) {
-    ser.begin_write_array(to_serdes_key(k));
+ARB_ARBOR_API void serialize(::arb::serializer& ser, const K& k, const std::vector<V, A>& vs) {
+    ser.begin_write_array(arb::to_serdes_key(k));
+    for (std::size_t ix = 0; ix < vs.size(); ++ix) serialize(ser, ix, vs[ix]);
+    ser.end_write_array();
+}
+
+template <typename K,
+          typename V>
+ARB_ARBOR_API void serialize(::arb::serializer& ser, const K& k, const std::vector<V>& vs) {
+    ser.begin_write_array(arb::to_serdes_key(k));
```

Comment:
Yeah, the second one can go. We don't want to serialize allocators since upon deserialization we want to use the current state, not the serialized one. Assume
```
auto vec = vec_w_alloc();
serialize(ser, vec); // Writes out allocator state
auto vec_two = vec_w_alloc(); // Change allocator state
deserialize(ser, vec); // Restores global(!!!) allocator state
// Now what happens?
auto vec_three = vec_w_alloc(); // Change allocator state
```